### ISSUE TITLE
[2016 ICLR] Unsupervised Representation Learning with Deep Convolutional Generative Adversarial Networks

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#23]
DCGAN을 제안한 논문

해당 논문은 4가지 contribution이 있으며, 이를 위주로 요약함

1. 모델 구조의 개선을 통해 train할 때 unstable한 점을 고침
2. 학습한 discriminator를 image classification에 적용했을 때, 잘됨을 보여줌 (compared to other unsupervised algorithms)
3. DCGAN의 학습된 특정 filter가 specific object를 generation하는 것을 보여줌
4. Generator에서 vector arithmetic property가 있음을 보여줌

각 1, 2, 3, 4가 주는 의미는 다음과 같음
1. 좋은 모델임
2. DCGAN의 discriminator가 잘 학습함
3. DCGAN의 generator가 image를 memorize하지 않고 generation함
4. DCGAN의 generator가 representation 능력이 좋음

### 1.1. Model
![image](https://user-images.githubusercontent.com/49643709/120126516-7ccfad00-c1f7-11eb-9251-33723e33a2c3.png)

기존의 CNN architecture를 사용하는 GAN은 scale이 큰 image generation에서 학습이 불안했음

저자들은 다양한 모델 exploration을 통해 stable하며 higher resolution image를 만들 수 있는 모델을 찾음

해당 모델을 찾는데 크게 4가지 요소가 반영됨

1. All convolutional net : pooling operation대신 convolution을 이용하여 upsampling, downsampling을 학습
2. Eliminate fully connected layers
3. Batch Normalization : poor initialization으로 인한 학습 문제와 deep models에서 gradient가 잘 전달될 수 있게 도와줌
4. ReLU & Leaky ReLU

### 1.2. Train
- Pre-processing : normalization [-1, 1]
- Mini-batch size 128, SGD with 0.0002 learning rate, 0.5 momentum
- Weight initialization : zero-centered Normal distribution with std 0.02
- Leaky ReLU : 0.2

## 2. Classification using GAN

Unsupervised representation learning algorithm의 quality를 평가하는 방법으로 supervised dataset의 feature extractor로 사용하여 performance를 측정함

저자들은 CIFAR-10, SVHN에서 실험하였음

### 2.1. CIFAR-10
![image](https://user-images.githubusercontent.com/49643709/120129032-3af63500-c1fe-11eb-9edb-2769949cbe74.png)

Feature extractor를 고정시킨 후, linear model만 fitting하여 classification score를 통해 평가

Imagenet-1k에서 학습한 DCGAN의 discriminator + L2 SVM을 이용하여 CIFAR-10에서 적용한 결과, feature를 잘 뽑아낸다는 것을 확인

### 2.2. SVHN

위와 동일한 방법으로 실행하여 그 당시의 sota를 찍음

또한 model performance의 key contributing factor가 CNN architecture가 아님을 보이기 위해 DCGAN의 discriminator model로 scratch부터 supervised learning으로 학습한 결과가 더 좋지 않음을 보여줌

## 3. Manipulating the generator representation

DCGAN 모델이 generation을 하는데 이걸 외워서 하지 않는지, major scene component object representation을 학습하는지 확인해야함

### 3.1. Not memorization, but learned
![image](https://user-images.githubusercontent.com/49643709/120136183-c7a7ef80-c20c-11eb-883f-b305e0104265.png)

GAN이 학습한 결과물들이 memorization이 아님을 보이기 위해, latent space에 변화를 주어 sharp transition이 있는지 확인

Sharp transition이 있다는 의미는 memorization을 했다고 볼 수 있음

Random point Z에서 interpolation을 한 결과, 이러한 현상은 찾아볼 수 없었으며 memorization이 아닌 학습함을 알 수 있음

### 3.2. Forgetting to draw certain object
![image](https://user-images.githubusercontent.com/49643709/120137850-30449b80-c210-11eb-8b0b-c5a468856700.png)

Major scene component object representation을 잘 학습하는지 확인하기 위하여 이미지 내의 window를 지우는 실험을 진행

실험은 다음과 같이 진행됨

1. Dataset 내의 image를 sample하여 window bounding box를 직접 그림
2. 2번째 convolution layer의 features에서 logistic regression을 이용하여 해당 feature activation이 window인지 아닌지 predict하도록 학습 (bounding box 내에 있으면 positive, 그 외에는 negative)
3. Logistic regression의 결과가 positive이면 해당 feature map을 drop
4. Dropped feature map을 이용하여 generation

해당 실험을 통해 잘 학습함을 확인할 수 있음

## 4. Vector arithmetic property
![image](https://user-images.githubusercontent.com/49643709/120138303-325b2a00-c211-11eb-93a7-bed080d4201d.png)

NLP에서 word의 learned representation이 잘됐는지 확인하는 방법으로 arithmetic operation을 이용하여 representation space에서 linear structure을 보였음

예시로 vector('KING') - vector('MAN') + vector('WOMAN')을 하면 가장 가까운 벡터는 vector('QUEEN')이 됨

이를 적용하여 Z('Man with glasses') - Z('Man without glasses') + Z('Woman without sunglasses')로 만들어진 Z로 generation을 하면 woman with glasses image가 생성됨을 보여줌

Single sample로 할 경우, 다소 불안정하기에 3개의 Z vector를 average하여 사용하였다고 함

추가적으로 이러한 arithmetic property가 image내에서 적용되지 않음을 보여줌
